design/recovery-internals.md

```diff
@@ -0,0 +1,142 @@
+# FDB Recovery Internals
+
+FDB uses recovery to handle various failures, such as hardware and network failures. When the current transaction system no longer works properly due to failures, recovery is automatically triggered to create a new generation of the transaction system. 
+
+This document explains at the high level how the recovery works in a single cluster. The audience of this document includes both FDB developers who want to have a basic understanding of the recovery process and database administrators who need to understand why a cluster fails to recover. This document does not discuss the complexity introduced to the recovery process by the multi-region configuration.
+
+
+## Overview
+
+Recovery is triggered on the cluster controller (CC). Coordinators choose a CC through leader election. If the current elected CC does not respond, a new CC will be elected by coordinators. If the majority of coordinators cannot be reached, CC will not be selected, recovery will get stuck.
```

Comment:
We should mention the recovery is about transaction system, which include master, proxy servers, transaction logs, and resolvers. Recovery doesn't involve storage servers.